### PR TITLE
Analyse 3D/2D array texture sampling data when creating context

### PR DESCRIPTION
**Pull request by Jupeyy**
Will create a texture, samples over it

test

and checks the pixel data.

I'm not the hugest fan of this hack, but if it works its fine

### CODE REVIEW

**Comment by Jupeyy** [comment]
@def- the only question is, how can we test this, the only person having this problem is the steam guy

**Comment by def-** [comment]
Should I send him a link to the Win64 download?

**Comment by Jupeyy** [comment]
Would work, but better prepare win32 too xD, so many ppl using 32bit, its insane :D

**Comment by def-** [comment]
Posted it here, let's see if we get a response: https://steamcommunity.com/app/412220/discussions/0/2923353981917186378/

**Comment by heinrich5991** [comment]
What should we do now that we don't get a response? Wait for someone with the same problem?

**Comment by Jupeyy** [comment]
Just release a version and see if he complains again xd,
I really dunno, if its just a sampler issue, it should be fixed.
If its related to that specific shader, its still not fixed.
Basically 50:50 chance.
but since something is rendered(even correctly, just wrongly displayed) the chance is higher for a sampler issue i think

**Comment by def-** [comment]
So let's get this in and try it out I guess? Agreed, @heinrich5991 ?

File: src/engine/client/backend_sdl.cpp [hunk 1]
```diff
@@ -1080,6 +1080,267 @@ void CCommandProcessorFragment_OpenGL2::SetState(const CCommandBuffer::SState &S
...
+
+		float m[2 * 4] = {
+			1, 0, 0, 0,
+			0, 1, 0, 0
+		};
+
+		// transpose bcs of column-major order of opengl
+		glUniformMatrix4x2fv(pProgram->m_LocPos, 1, true, (float*)&m);
+	}
+	else {
```
Comment (heinrich5991) [inline src/engine/client/backend_sdl.cpp:1195]:
```suggestion
	else
	{
```

**Review by heinrich5991 — COMMENTED** [review]
LGTM otherwise.

**Comment by Jupeyy** [comment]
this is already merged btw @heinrich5991, deen cherry picked to test the client

**Comment by Jupeyy** [comment]
I'll close this for now, when there is an issue, its better to create a new issue, because there are alot of commits now that are built upon the files here anyway.

**Pull request closed without merging**